XamlAnimatedGif.WinRT/NetworkHelper.cs

```diff
@@ -0,0 +1,57 @@
+﻿#region
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Windows.Security.Cryptography;
+using Windows.Security.Cryptography.Core;
+using XamlAnimatedGif.Interfaces;
+
+#endregion
+
+namespace XamlAnimatedGif
+{
+    internal class NetworkHelper : INetworkHelper
+    {
+        private readonly ICacheHelper _cacheHelper;
+
+        public NetworkHelper()
+        {
+            _cacheHelper = new CacheHelper();
+        }
+
+        public async Task<Stream> GetNetworkStreamAsync(Uri uri)
+        {
+            //generating temp file name by hashing the url
+            var tempId = GetHash(HashAlgorithmNames.Sha1, uri.AbsoluteUri);
+            var cacheStream = await _cacheHelper.GetAsync(tempId);
+
+            if (cacheStream != null)
+                return cacheStream;
+
+            //no cache, continue with download
+            using (var client = new HttpClient())
+            {
+                //fails if the status is not a success one
+                var bytes = await client.GetByteArrayAsync(uri);
+                //using a memory stream, need a seekable one
+                var mem = new MemoryStream(bytes);
+
+                //cache the gif
+                await _cacheHelper.SaveAsync(tempId, mem);
+
+                return mem;
```

Comment:
This causes the entire file to be kept in memory; we don't want that... keeping the memory usage low is one of the most important principles for this library (in contrast with its predecessor, WpfAnimatedGif). We should always return a file stream from this method, and the data should be downloaded directly to the file, not using an intermediary byte array.